src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/dto/BambooBuildResultNotificationDTO.java

```diff
@@ -49,6 +55,48 @@ public void setBuild(BambooBuildDTO build) {
         this.build = build;
     }
 
+    @Override
+    public ZonedDateTime getBuildRunDate() {
+        return getBuild().getBuildCompletedDate();
+    }
+
+    @Override
+    public Optional<String> getCommitHashFromAssignmentRepo() {
+        return getCommitHashFromRepo(ASSIGNMENT_REPO_NAME);
+    }
+
+    @Override
+    public Optional<String> getCommitHashFromTestsRepo() {
+        return getCommitHashFromRepo(TEST_REPO_NAME);
+    }
+
+    @Override
+    public boolean isBuildSuccessful() {
+        return getBuild().isSuccessful();
+    }
+
+    @Override
+    public Long getBuildScore() {
+        // the real score is calculated in the grading service
+        return 0L;
+    }
+
+    @Override
+    public String getTestsPassedString() {
+        if (getBuild().getTestSummary().getDescription().equals("No tests found")) {
```

Comment:
```suggestion
        if ("No tests found".equals(getBuild().getTestSummary().getDescription())) {
```

Reply:
I think that change makes it even less readable

Reply:
yes, but this is the recommended way, to avoid possible null pointer exceptions

Reply:
There is also Objects.equals()

Reply:
So lets just it like this?